pkg/controller/eventing/inmemory/clusterchannelprovisioner/reconcile.go

```diff
@@ -96,12 +103,17 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	err = r.reconcile(ctx, ccp)
 	if err != nil {
 		logger.Info("Error reconciling ClusterChannelProvisioner", zap.Error(err))
+		r.recorder.Eventf(ccp, corev1.EventTypeWarning, ccpReconcileFailed, "Failed to reconcile ClusterChannelProvisioner: %v", err)
```

Comment:
Is it possible to get here without already having another event emitted? If not, then I think we can remove this event.